addons/website_quote/controllers/main.py

```diff
@@ -40,7 +40,9 @@ def view(self, order_id, pdf=None, token=None, message=False, **post):
             pdfhttpheaders = [('Content-Type', 'application/pdf'), ('Content-Length', len(pdf))]
             return request.make_response(pdf, headers=pdfhttpheaders)
         user = request.registry['res.users'].browse(request.cr, SUPERUSER_ID, request.uid, context=request.context)
-        tx_id = request.registry['payment.transaction'].search(request.cr, SUPERUSER_ID, [('reference', '=', order.name)], context=request.context)
+        tx_id = request.session.get('quote_transaction_id')
+        if not tx_id:
+            tx_id = request.registry['payment.transaction'].search(request.cr, SUPERUSER_ID, [('reference', '=', order.name)], context=request.context)
```

Comment:
this will only work if there hasn't been any "SO145325-2" integer appeding; maybe a like would be preferable